llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4858,13 +4858,15 @@ static Value *simplifySelectInst(Value *Cond, Value *TrueVal, Value *FalseVal,
   // select ?, undef,  X -> X
   if (isa<PoisonValue>(TrueVal) ||
       (Q.isUndefValue(TrueVal) &&
-       isGuaranteedNotToBePoison(FalseVal, Q.AC, Q.CxtI, Q.DT)))
+       (isGuaranteedNotToBePoison(FalseVal, Q.AC, Q.CxtI, Q.DT) ||
```

Comment:
Can you drop the isGuaranteedNotToBePoison call? Or does the loss of context-sensitivity cause regressions?

Reply:
I think it is safe to drop the call. Loss of context-sensitivity doesn't cause regressions (in both llvm regression tests and my benchmark).